content/200-orm/200-prisma-client/400-type-safety/100-operating-against-partial-structures-of-model-types.mdx

```diff
@@ -128,10 +128,22 @@ const usersWithPosts: UsersWithPosts = await getUsersWithPosts()
 
 ### Solution
 
-With the `PromiseReturnType` that is exposed by the `Prisma` namespace, you can solve this more elegantly:
+You can use native the TypeScript utility type [`Awaited`](https://www.typescriptlang.org/docs/handbook/utility-types.html#awaitedtype) and [`ReturnType`](https://www.typescriptlang.org/docs/handbook/utility-types.html#returntypetype) to solve the problem elegantly:
+
+```ts
+type UsersWithPosts = Prisma.Awaited<ReturnType<typeof getUsersWithPosts>>
+```
+
+When using the `prisma-client-js` generator, a `PromiseReturnType` is exposed by the `Prisma` namespace, you can solve the problem using the `PromiseReturnType` type:
```

Comment:
I am not convinced we still need to mention that `PromiseReturnType` exists at all. I would just rewrite this section to use `Awaited<ReturnType<typeof getUsersWithPosts>>` and that's it. Mentioning both seems redundant and potentially confusing for users to me.